weldx/core.py

```diff
@@ -980,7 +975,7 @@ def __init__(
             if dims is not None and not isinstance(dims, list):
                 raise ValueError(f"Argument 'dims' must be list of strings, not {dims}")
             self._init_discrete(obj, dims, coords)
-        elif isinstance(obj, (MathematicalExpression, str)):
+        elif isinstance(obj, (MathematicalExpression, str, sympy.Expr)):
```

Comment:
This can be a "typedef", right?

Reply:
While this is a good idea, the `MathematicalExpression` makes it prone for circular imports. At least when I tried it, those occurred.

Reply:
you could also just write it as a str in the typedef, if it causes a circular import. The type annotation system is going to be string only in future Python versions, right?